newIDE/app/public/external/piskel/piskel-main.js

```diff
@@ -1,255 +1,127 @@
-    'fake-piskelapp-header'
-  )[0];
-
-  saveFolderLabel = editorContentDocument.createElement('label');
-  saveFolderLabel.addEventListener('click', selectBaseFolderPath);
-  saveFolderLabel.style = headerStyle.saveFolderLabel;
-  piskelAppHeader.appendChild(saveFolderLabel);
-
-  piskelAnimationNameInput = editorContentDocument.createElement('input');
-  piskelOptions = { name: 'New Animation' };
-  piskelAnimationNameInput.oninput = updatePiskelBasePath;
-  piskelAnimationNameInput.type = 'text';
-  piskelAnimationNameInput.style = headerStyle.piskelAnimationNameInput;
-  piskelAppHeader.appendChild(piskelAnimationNameInput);
-
-  const saveButton = editorContentDocument.createElement('button');
-  saveButton.textContent = 'Save to GDevelop';
-  saveButton.style = headerStyle.saveButton;
-  piskelAppHeader.appendChild(saveButton);
-  saveButton.addEventListener('click', saveToGD);
-
-  const cancelButton = editorContentDocument.createElement('button');
-  cancelButton.textContent = 'Cancel';
-  cancelButton.style = headerStyle.cancelButton;
-  piskelAppHeader.appendChild(cancelButton);
-  cancelButton.addEventListener('click', cancelChanges);
-
-  setFolderButton = editorContentDocument.createElement('button');
-  setFolderButton.textContent = 'Set Folder';
-  setFolderButton.style = headerStyle.setFolderButton;
-  piskelAppHeader.appendChild(setFolderButton);
-  setFolderButton.addEventListener('click', selectBaseFolderPath);
-};
-
-function fileExists(path) {
-  try {
-    return fs.statSync(path).isFile();
-  } catch (e) {
-    if (e.code === 'ENOENT') {
-      // no such file or directory. File really does not exist
-      return false;
-    }
-    console.log('Exception fs.statSync (' + path + '): ' + e);
-    throw e; // something else went wrong, we don't have rights, ...
-  }
-}
-
-/**
- * Returns a path for a file that does not exist yet.
- * Used to avoid unwanted file overwriting.
- */
-function makeFileNameUnique(path) {
-  if (!fileExists(path)) {
-    return path;
-  }
-
-  //TODO: refactor by using `path` module and properly handle the case
-  //where the file has no extension.
-  const folderPath = path.substring(0, path.lastIndexOf('/') + 1);
-  const extension = path.substring(path.lastIndexOf('.'), path.length);
-  const oldFileName = path.substring(
-    path.lastIndexOf('/') + 1,
-    path.lastIndexOf('.')
-  );
-  let appendNumber = 0;
-  let newUniqueNamePath =
-    folderPath + oldFileName + '-' + String(appendNumber) + extension;
-  while (fileExists(newUniqueNamePath)) {
-    appendNumber += 1;
-    newUniqueNamePath =
-      folderPath + oldFileName + '-' + String(appendNumber) + extension;
+// Repeatedly try to gain access to piskel's control element and its methods
+// When succeeding, stop trying.
+let pskl = null;
+const tryToGetPiskel = () => {
+  if (pskl === null) {
+    pskl = document.querySelector('#piskel-frame').contentWindow.pskl;
+  } else {
+    // gained access to control elements!
```

Comment:
You can remove this comment, I think it's obvious by the function name.

also, this should be:
```
const tryToGetPiskel = () => {
  pskl = document.querySelector('#piskel-frame').contentWindow.pskl;
  if (pskel !== null) {
    ipcRenderer.send('piskel-ready');
    clearInterval(retryToGetPiskel);
  }
}
```

Otherwise, you're "wasting" an interval, because when piskel is ready, you'll have to wait for the next interval before seeing that pskl is not null anymore.